MapboxDirections/MBIntersection.swift

```diff
@@ -59,30 +60,30 @@ public class Intersection: NSObject, NSSecureCoding {
     }
     
     internal convenience init(json: JSONDictionary) {
-        let approachIndex = json["in"] as? Int
-        let outletIndex = json["out"] as? Int
+        let approachIndex = json["in"] as? Int ?? -1
+        let outletIndex = json["out"] as? Int ?? -1
         let entry = json["entry"] as! [Bool]
-        let locationArray = json["location"] as! [Double]
-        let location = CLLocationCoordinate2D(latitude: locationArray[0], longitude: locationArray[1])
+        let coords = json["location"] as! [Double]
+        let location = CLLocationCoordinate2D.init(geoJSON: coords)
         let headings = json["bearings"] as! [CLLocationDirection]
         let lanesJSON = json["lanes"] as? [JSONDictionary]
         var lanes = [Lane]()
         var usableLanes = Set<Lane>()
         
-        lanesJSON?.forEach({ (laneJSON) in
+        for laneJSON in lanesJSON ?? [] {
             let lane = Lane(json: laneJSON)
             lanes.append(lane)
             if laneJSON["valid"] as! Bool {
                 usableLanes.insert(lane)
             }
-        })
+        }
         
         self.init(approachIndex: approachIndex, outletIndex: outletIndex, entry: entry, location: location, headings: headings, lanes: lanes, usableLanes: usableLanes)
     }
     
     public required init?(coder decoder: NSCoder) {
-        approachIndex = decoder.decodeObjectForKey("approachIndex") as? Int
-        outletIndex = decoder.decodeObjectForKey("outletIndex") as? Int
+        approachIndex = decoder.decodeObjectForKey("approachIndex") as! Int
+        outletIndex = decoder.decodeObjectForKey("outletIndex") as! Int
```

Comment:
#81 correctly points out that we need to use `decodeObjectOfClass(_:forKey:)` to be secure. If #81 lands first, we should update the added classes in this PR to use `decodeObjectOfClass(_:forKey:)` too.